var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -193,16 +203,58 @@ def flag_handler(self, name, flags):
 
         return (flags, None, None)
 
-    @run_before("install")
-    def set_blas_lapack(self):
-        # https://numpy.org/devdocs/user/building.html
-        # https://github.com/numpy/numpy/blob/master/site.cfg.example
+    def blas_lapack_pkg_config(self) -> Tuple[str, str]:
+        """Convert library names to pkg-config names.
 
-        # Skip if no BLAS/LAPACK requested
+        Returns:
+            The names of the blas and lapack libs that pkg-config should search for.
+        """
         spec = self.spec
-        if "+blas" not in spec and "+lapack" not in spec:
-            return
+        blas = spec["blas"].libs.names[0]
+        lapack = spec["lapack"].libs.names[0]
+
+        if spec["blas"].name in ["intel-mkl", "intel-parallel-studio", "intel-oneapi-mkl"]:
+            blas = "mkl-dynamic-lp64-seq"
 
+        if spec["lapack"].name in ["intel-mkl", "intel-parallel-studio", "intel-oneapi-mkl"]:
+            lapack = "mkl-dynamic-lp64-seq"
+
+        if spec["blas"].name in ["blis", "amdblis"]:
+            blas = "blis"
+
+        if "armpl" in blas:
+            if "_mp" in blas:
+                blas = "armpl-dynamic-lp64-omp"
+            else:
+                blas = "armpl-dynamic-lp64-seq"
+
+        if "armpl" in lapack:
+            if "_mp" in lapack:
+                lapack = "armpl-dynamic-lp64-omp"
+            else:
+                lapack = "armpl-dynamic-lp64-seq"
+
+        return blas, lapack
+
+    @when("@1.26:")
+    def config_settings(self, spec, prefix):
```

Comment:
These are the first packages that actually use config_settings so we can still change the design of this if it makes sense. Maybe a nested dict would make more sense?

Reply:
I think a nested dict may be better indeed. This seems to work according to the CI logs, but I'm not sure if it always will. The spec for `config_settings` is a dict with string key-values, and the below looks potentially confusing. The actual key is `setup-args`, used twice, with values `f"-Dblas={blas}" and  `f"-Dlapack={lapack}". But you have to spell it here with a key value of `setup-args=-Dblas`.

Reply:
Done. This was actually a lot of fun, it's been a long time since I've written real code like this.